File: engine/core/painting/Picture.cpp
// Copyright 2015 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "sky/engine/core/painting/Picture.h"

#include "base/logging.h"

namespace blink {

PassRefPtr<Picture> Picture::create(PassRefPtr<SkPicture> skPicture)
{
    ASSERT(skPicture);
    return adoptRef(new Picture(skPicture));
}

Picture::Picture(PassRefPtr<SkPicture> skPicture)
    : m_picture(skPicture)
{
}

Picture::~Picture()
{
}

} // namespace blink

File: lib/web_ui/lib/src/engine/compositor/picture.dart
// Copyright 2013 The Flutter Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

part of engine;

class CkPicture implements ui.Picture {
  final SkiaObject skPicture;
  final ui.Rect? cullRect;

  CkPicture(this.skPicture, this.cullRect);

  @override
  int get approximateBytesUsed => 0;

  @override
  void dispose() {
    skPicture.delete();
  }

  @override
  Future<ui.Image> toImage(int width, int height) {
    throw UnsupportedError('Picture.toImage not yet implemented for CanvasKit and HTML');
  }
}
